gluten-core/src/main/scala/io/glutenproject/extension/columnar/TransformHintRule.scala

```diff
@@ -276,24 +276,32 @@ case class AddTransformHintRule() extends Rule[SparkPlan] {
           if (!enableColumnarBatchScan) {
             TransformHints.tagNotTransformable(plan)
           } else {
-            val transformer = new BatchScanExecTransformer(plan.output, plan.scan,
-              plan.runtimeFilters)
-            TransformHints.tag(plan, transformer.doValidate().toTransformHint)
+            if (plan.runtimeFilters.nonEmpty) {
+              TransformHints.tagTransformable(plan)
```

Comment:
why considered transformable here?

Reply:
Keep a plan to be transformable by default, like [this](https://github.com/oap-project/gluten/blob/095a637ce0bd395515a0bd007a44c45f271f6864/gluten-core/src/main/scala/io/glutenproject/extension/columnar/TransformHintRule.scala#L567). If marked "TRANSFORM_UNSUPPORTED", the plan won't reach [here](https://github.com/oap-project/gluten/blob/main/gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala#L120).

Reply:
Is it possible to do the validation only in this file, instead of [here](https://github.com/oap-project/gluten/blob/095a637ce0bd395515a0bd007a44c45f271f6864/gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala#L368-L400)?

By validation I mean the calls to `xxx.doValidate`, or the tagging actions on the plan.

Reply:
Generally, we should do the validation in TransformHintRule and replace the plan to complete the transformer in ColumnarOverrides. However, we need to transform some plan's inner operator, even when they are marked "TRANSFORM_UNSUPPORTED".